dashboard/test/models/contact_rollups_pardot_memory_test.rb

```diff
@@ -31,4 +31,54 @@ class ContactRollupsPardotMemoryTest < ActiveSupport::TestCase
 
     assert_equal new_pardot_id, ContactRollupsPardotMemory.find_by(email: existing_record.email)&.pardot_id
   end
+
+  test 'create_new_pardot_contacts' do
+    assert_equal 0, ContactRollupsPardotMemory.count
+    assert_equal 0, ContactRollupsProcessed.count
+    PardotV2.stubs(:submit_batch_request).returns([])
+
+    # Sync 1 contact
+    processed_contacts = []
+    processed_contacts << (create :contact_rollups_processed)
+
+    ContactRollupsPardotMemory.create_new_pardot_contacts
+    assert_equal processed_contacts.length, ContactRollupsPardotMemory.count
+
+    # Sync multiple contacts
+    processed_contacts.concat(create_list(:contact_rollups_processed, 2))
+
+    ContactRollupsPardotMemory.create_new_pardot_contacts
+    assert_equal processed_contacts.length, ContactRollupsPardotMemory.count
+    processed_contacts.each do |contact|
+      refute_nil ContactRollupsPardotMemory.find_by(email: contact.email)
+    end
+  end
+
+  test 'save_sync_results' do
+    assert_equal 0, ContactRollupsPardotMemory.count
+
+    submissions = [
+      {email: 'invalid_email', id: nil, db_Opt_in: 'No'},
+      {email: 'valid@domain.com', id: nil, db_Opt_in: 'Yes'},
+    ]
+    error_msg = 'Invalid prospect email address'
```

Comment:
Asking for my knowledge here...is it better to reference the constant or use the raw string in tests?

Reply:
It is better to use constant when possible. I'll change it to `PardotHelpers::ERROR_INVALID_EMAIL`